webhub/tests.py

```diff
@@ -1,8 +1,341 @@
+        self.assertEqual(len(post_list_before), len(post_list_after))
+
+        for post in post_list_before:
+            self.assertEqual(Post.objects.get(id=post.id), post)
+
+    def test_detail_unauthenticated_cases(self):
+        """Test unauthenticated API calls to post-detail endpoint"""
+
+        self.unauthenticate()
+        post_list_before = Post.objects.all().order_by('id')
+
+        for post in post_list_before:
+
+            post_id = str(post.id)
+            url = reverse('post-detail', args=[post_id])
+            response = self.client.delete(url)
+            self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+            response = self.client.get(url)
+            self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+            response = self.client.head(url)
+            self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+            response = self.client.options(url)
+            self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+            response = self.client.post(url, self.data_1, format='json')
+            self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+            response = self.client.put(url, self.data_1, format='json')
+            self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        post_list_after = Post.objects.all().order_by('id')
+        self.assertEqual(len(post_list_before), len(post_list_after))
+
+        for post in post_list_before:
+            self.assertEqual(Post.objects.get(id=post.id), post)
+
+    def test_list_delete_cases(self):
+        """Test HTTP DELETE API calls to post-list endpoint"""
+
+        url = reverse('post-list')
+        post_list = Post.objects.all().order_by('id')
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code,
+                         status.HTTP_405_METHOD_NOT_ALLOWED)
+
+        for post in post_list:
+            self.assertIsNotNone(Post.objects.get(id=post.id))
+
+    def test_list_get_cases(self):
+        """Test HTTP GET API calls to post-list endpoint"""
+
+        url = reverse('post-list')
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        post_list = Post.objects.all().order_by('id')
+        stream = BytesIO(response.render().content)
+        # parse JSON in Python native datatype (dictionary)
+        # so we can iterate over the result
+        data = JSONParser().parse(stream)
+        results = data['results']
+        i = 0
+
+        for post in results:
+            # compare JSON objects
+            serializer = PostSerializer(post_list[i])
+            content_db = JSONRenderer().render(serializer.data)
+            serializer = PostSerializer(post)
+            content_api = JSONRenderer().render(serializer.data)
+            # assert is failing because PostSerializer sets
```

Comment:
Yes, an assertion is missing here because I need to fix the PostSerializer first. I will add the assert back in when I've fixed the PostSerializer.